smolpickle.c

```diff
@@ -3468,6 +3522,49 @@ load_buildstruct(UnpicklerObject *self)
     return res;
 }
 
+static int
+load_enum(UnpicklerObject *self)
+{
+    PyObject *val = NULL, *code = NULL, *typ = NULL, *obj;
+    SmolpickleState *st;
+    int res = -1;
+
+    STACK_POP(self, val);
+    if (val == NULL)
+        goto cleanup;
+
+    STACK_POP(self, code);
+    if (code == NULL)
+        goto cleanup;
+    if (self->registry != NULL) {
+        typ = PyDict_GetItem(self->registry, code);
+    }
+    if (typ == NULL) {
+        PyErr_Format(PyExc_ValueError,
+                     "Typecode %R isn't in type registry",
+                     code);
+        goto cleanup;
+    }
+    st = smolpickle_get_global_state();
+    if (!PyObject_IsSubclass(typ, st->EnumType)) {
```

Comment:
If we check and convert `st->EnumType` to a `PyTypeObject` on module init, we can use `PyType_IsSubtype` here instead, which would be more efficient and avoids accidental invocation of `__subclasscheck__`.

Reply:
We might also drop the typecheck here, not sure.